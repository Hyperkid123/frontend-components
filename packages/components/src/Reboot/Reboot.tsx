import React from 'react';
import classNames from 'classnames';
import RebootingIcon from '@patternfly/react-icons/dist/dynamic/icons/rebooting-icon';

import './reboot.scss';

export interface RebootProps extends React.DetailedHTMLProps<React.HTMLAttributes<HTMLSpanElement>, HTMLSpanElement> {
  className?: string;
  red?: boolean;
}

const Reboot: React.FC<RebootProps> = ({ red, className, ...props }) => {
  const rebootIconClasses = classNames('ins-c-reboot', { [`ins-m-red`]: red }, className);

  return (
    <span className={rebootIconClasses} {...props}>
      <RebootingIcon />
      <span>Reboot required</span>
    </span>
  );
};

export default Reboot;
