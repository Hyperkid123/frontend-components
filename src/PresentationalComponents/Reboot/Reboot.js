import React from 'react';
import propTypes from 'prop-types';

import classNames from 'classnames';

import { RebootingIcon } from '@patternfly/react-icons';

import './reboot.scss';

/**
 * This is a dumb component that only recieves properties from a smart component.
 * Dumb components are usually functions and not classes.
 *
 * @param props the props given by the smart component.
 */

const Reboot = ({ red, className, ...props }) => {

    const rebootIconClasses = classNames(
        'ins-c-reboot',
        { [`ins-m-red`]: red },
        className
    );

    return (
        <span className={ rebootIconClasses } { ...props }>
            <RebootingIcon/>
            <span>Needs Reboot</span>
        </span>
    );
};

export default Reboot;

Reboot.propTypes = {
    className: propTypes.string,
    red: propTypes.bool
};
