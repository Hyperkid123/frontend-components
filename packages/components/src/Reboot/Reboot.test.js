import React from 'react';
import Reboot from './Reboot';
import { render } from '@testing-library/react';

describe('Reboot component', () => {
  it('should render correctly', () => {
    const { container } = render(<Reboot />);
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with red', () => {
    const { container } = render(<Reboot red />);
    expect(container).toMatchSnapshot();
  });
});
