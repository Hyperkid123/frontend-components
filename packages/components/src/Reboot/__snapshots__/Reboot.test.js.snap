// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reboot component should render correctly 1`] = `
<span
  className="ins-c-reboot"
>
  <RebootingIcon />
  <span>
    Reboot required
  </span>
</span>
`;

exports[`Reboot component should render correctly with red 1`] = `
<span
  className="ins-c-reboot ins-m-red"
>
  <RebootingIcon />
  <span>
    Reboot required
  </span>
</span>
`;
